chamber/middleware.py

```diff
@@ -0,0 +1,34 @@
+from django.db import transaction
+from django.db import connections
+from django.urls import get_resolver
+
+from chamber.utils.transaction import transaction_signals
+
+
+def get_view_from_request_or_none(request):
+    try:
+        if hasattr(request, 'urlconf'):
+            urlconf = request.urlconf
+            set_urlconf(urlconf)
+            resolver = get_resolver(urlconf)
+        else:
+            resolver = get_resolver()
+        return resolver.resolve(request.path_info)
+    except Resolver404:
+        return None
+
+
+class TransactionSignalsMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        non_atomic_requests = getattr(get_view_from_request_or_none(request), '_non_atomic_requests', set())
+        for db in connections.all():
+            if db.settings_dict['CHAMBER_ATOMIC_REQUESTS'] and db.alias not in non_atomic_requests:
```

Comment:
Maybe use `settings_dict.get()`, or you really want to enforce that this setting is explicit?

Reply:
good point